src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -273,7 +273,7 @@ public override void EnsureRenderedBodyOrSections()
             }
         }
 
-        public override void BeginContext(int position, int length, bool isLiteral)
+        protected override void BeginContextImpl(int position, int length, bool isLiteral)
```

Comment:
I'd really prefer another name other than `Impl` if it's public API. How about `BeginContextCore`